airsonic-main/src/main/java/org/airsonic/player/service/MediaScannerService.java

```diff
@@ -124,16 +142,26 @@ public synchronized void schedule() {
     /**
      * Returns whether the media library is currently being scanned.
      */
-    public synchronized boolean isScanning() {
+    public boolean isScanning() {
         return scanning;
     }
 
     /**
      * Returns the number of files scanned so far.
      */
     public int getScanCount() {
-        return scanCount;
+        return scanCount.get();
     }
+    
+    private ForkJoinWorkerThreadFactory mediaScannerThreadFactory = new ForkJoinWorkerThreadFactory() {
+        @Override           
+        public ForkJoinWorkerThread newThread(ForkJoinPool pool) {
+            final ForkJoinWorkerThread worker = ForkJoinPool.defaultForkJoinWorkerThreadFactory.newThread(pool);
+            worker.setName("MediaLibraryScanner-" + worker.getPoolIndex());
+            worker.setPriority(Thread.MIN_PRIORITY);
+            return worker;
+        }
+    };
```

Comment:
The fork join thread factory for our pool. Just sets the name (so it's identifiable via debugging) and priority (we can consider making this configurable).